README.md

```diff
@@ -1,40 +1,96 @@
+* `check-directory.js` - custom function to check if a directory exists and if not create it, used in outputting eslint error logs
+* `common-paths.js` - contains base path variables to be composed from in build-paths.js and dist-paths.js
+* `concat.js` - contains object with two Gulp concat task functions, one with and one without gulp-newer
+* `custom-plumber.js` - error notification function using gulp-notify
+* `dist-names.js` - contains variable names of files outputted into the build folder
+* `dist-paths.js` - contains path variables of files going that are being outputted into the `/assets/build` folder by Gulp
+* `gulp-load-plugins.js` - contains function to pull in `package.json` plugins using gulp-load-plugins and store it in the `$` variable. Scoped to just devDependencies plugins.
+*  `uglify.js` - contains function to run gulp-uglify task, preserves comments that starts with @license or @preserve, also shows anticipated gzipped size 
+
+
+### Gulp Processing for SCSS, JS, CSS files
+ * SASS files - The src `_*.scss` partials are linted and then copied to the `assets/dist/scss` folder. Lint errors will be displayed in the console and also outputed to `/logs/stylelint` 
+ * JS files:
+    * src vendor JS files are concatenated and uglified and outputted to the `assets/dist/js/vendor` folder with sourcemap.
+    * src mustard JS files are concatenated and uglified and outputted to the `assets/dist/js/mustard` folder with sourcemap.
+    * src app/common JS files are linted, transpiled, concatenated and uglified and outputted to the `assets/dist/js/app/common` folder with sourcemap. Lint errors will be displayed in the console and outputted to `/logs/eslint`.
+    * src app/optional JS files are linted and copied to `assets/dist/js/app/optional` folder. Lint errors will be displayed in the console and outputted to `/logs/eslint`.
+* CSS files - The src css files are copied to the `assets/dist/css` folder
+    
+    
+
+## Installing the project
+
+Run `npm install` to install the required node modules.
+You are good to go!
+
+## Running Gulp Tasks - Commands
+* `gulp` - the default gulp task. The one task to rule them all. Deletes `/build` and `/dist` and runs all the SCSS, JS, CSS related tasks once and enters into watching mode
+* `gulp stylelintFix` - stylelint allows _some_ CSS/SCSS styling errors to be fixed automatically. It is not perfect but definitely helps. Run this task manually to have stylelint automatically fix some of the minor styling issues in your SCSS.
+* `gulp eslintFix` - eslint allows _some_ JS errors to be fixed automatically. It is not perfect but definitely helps. Does convert JS to ES6 version. Run this task manually to have eslint automatically fix some of the minor styling issues in your JS.
 
-Run the following command to build the production files:
-```
-npm run dev -- --env.theme your_theme_name
-```
+## Known issues
```

Comment:
Just want to say thank you for documenting all of this! 🥇

Reply:
I enjoyed pouring my thoughts out 👍